rules/error-handling.md

```diff
@@ -0,0 +1,1060 @@
+        
+        // Validate categories
+        $categories = $request->get_param('categories');
+        if (!empty($categories) && is_array($categories)) {
+            foreach ($categories as $category_id) {
+                if (!term_exists($category_id, 'category')) {
+                    $errors->add(
+                        'invalid_category',
+                        sprintf(
+                            /* translators: %d: Category ID */
+                            __('Category with ID %d does not exist.', 'textdomain'),
+                            $category_id
+                        ),
+                        ['field' => 'categories', 'invalid_id' => $category_id]
+                    );
+                }
+            }
+        }
+        
+        // Return errors if any exist
+        if ($errors->has_errors()) {
+            return $errors;
+        }
+        
+        return true;
+    }
+    
+    /**
+     * Validate business rules.
+     *
+     * @param WP_REST_Request $request Request object.
+     * @return WP_Error|true WP_Error if validation fails, true on success.
+     */
+    private function validate_business_rules(WP_REST_Request $request) {
+        $errors = new WP_Error();
+        
+        // Check user permissions
+        if (!current_user_can('publish_posts') && 'publish' === $request->get_param('status')) {
+            $errors->add(
+                'insufficient_permissions',
+                __('You do not have permission to publish posts.', 'textdomain'),
+                ['status' => 403]
+            );
+        }
+        
+        // Check for duplicate content
+        $title = $request->get_param('title');
+        if (!empty($title)) {
+            $existing_post = get_page_by_title($title, OBJECT, 'post');
+            if ($existing_post) {
+                $errors->add(
+                    'duplicate_title',
+                    __('A post with this title already exists.', 'textdomain'),
+                    ['field' => 'title', 'existing_post_id' => $existing_post->ID]
+                );
+            }
+        }
+        
+        return $errors->has_errors() ? $errors : true;
+    }
+}
+```
+
+### AJAX Error Handling
+
+```php
+<?php
+/**
+ * AJAX handler with WP_Error for form processing.
+ */
+class Contact_Form_Handler {
+    
+    /**
+     * Initialize AJAX hooks.
+     */
+    public function init(): void {
+        add_action('wp_ajax_submit_contact_form', [$this, 'handle_contact_form']);
+        add_action('wp_ajax_nopriv_submit_contact_form', [$this, 'handle_contact_form']);
+    }
+    
+    /**
+     * Handle contact form submission.
+     */
+    public function handle_contact_form(): void {
+        // Verify nonce
+        if (!wp_verify_nonce($_POST['nonce'] ?? '', 'contact_form_nonce')) {
+            wp_send_json_error([
+                'message' => __('Security verification failed.', 'textdomain'),
+                'code' => 'invalid_nonce'
+            ]);
```

Comment:
Remove all examples using wp_send_json_error that are not using WP_Error instances